src/participant.rs

```diff
@@ -168,20 +169,23 @@ where
     }
 }
 
-pub(crate) trait ProtocolParticipant {
-    /// Input type used to process incoming messages.
-    type Input;
+/// These are the public-facing methods that must be implemented for a given
+/// protocol.
+pub trait ProtocolParticipant {
+    /// Input type for a new protocol instance.
+    type Input: Debug;
     /// Output type of a successful protocol execution.
     type Output: Debug;
 
-    /// Returns a reference to the [`LocalStorage`] associated with this
-    /// protocol.
-    fn local_storage(&self) -> &LocalStorage;
-    /// Returns a mutable reference to the [`LocalStorage`] associated with this
-    /// protocol.
-    fn local_storage_mut(&mut self) -> &mut LocalStorage;
-    fn id(&self) -> ParticipantIdentifier;
-    fn other_ids(&self) -> &Vec<ParticipantIdentifier>;
+    /// The `MessageType` of a "ready" message, signalling that a participant is
```

Comment:
This is weird, actually. `MessageType` is a public type in a private module, so it's totally opaque to the user, but apparently the public trait can still return it. I rephrased the documentation to not reference the type itself (because the link breaks if I add brackets).